File: 217-ContainsDuplicate/217-ContainsDuplicate.java
// Last updated: 5/21/2026, 12:06:20 AM
import java.util.Arrays;
import java.util.Set;
import java.util.stream.Collectors;

class Solution {
    public boolean containsDuplicate(int[] nums) {
        Set<Integer> set = Arrays.stream(nums).boxed().collect(Collectors.toSet());
        return !(nums.length == set.size());
    }
}

File: 217-ContainsDuplicate/217-ContainsDuplicate.cpp
// Last updated: 5/21/2026, 12:06:19 AM
#include <unordered_set>
#include <vector>

class Solution {
public:
    bool containsDuplicate(std::vector<int>& nums) {
        std::unordered_set<int> set;
        
        for (const auto& num : nums) {
            if (set.count(num))
                return true;
            set.insert(num);
        }
        
        return false;
    }
};
